packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -488,6 +495,66 @@ function compileProofNumericValues(
   };
 }
 
+export function compileCommonEntryInequalities<
+  CompilerEntryInfo extends { entryConfig: GPCProofEntryConfig }
+>(
+  boundsCheckConfig: GPCProofBoundsCheckConfig,
+  entryMap: Map<PODEntryIdentifier, CompilerEntryInfo>,
+  paramEntryInequalities: number
+): {
+  entryInequalityValueIndex: CircuitSignal[];
+  entryInequalityOtherValueIndex: CircuitSignal[];
+  entryInequalityIsLessThan: CircuitSignal;
+} {
+  const numericValueIndex: Record<PODEntryIdentifier, bigint> =
+    Object.fromEntries(
+      Object.keys(boundsCheckConfig).map((entryIdentifier, i) => [
+        entryIdentifier,
+        BigInt(i)
+      ])
+    );
+  const signalTriples = (
```

Comment:
This deserves a big comment explaining what's going on.  I found it very hard to understand what these triples are through the details of the flatmap.  The concept of what you're doing, though, is quite simple, where you're turning the 4 different types of comparison in config into the input order and output value of the 1 type of module we use.